specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/LinkedService.json

```diff
@@ -1243,16 +1243,20 @@
     "Db2LinkedServiceTypeProperties": {
       "description": "DB2 linked service properties.",
       "properties": {
+        "connectionString": {
```

Comment:
Adding new properties to the response of stable APIs constitutes as a breaking change. This doc explains why:
https://microsoft.sharepoint.com/:w:/r/teams/azureresourcemanagerteam/_layouts/15/guestaccess.aspx?e=xEcppo&share=EWXsAQ1yx25KkyYCeeWGUgwBSBxEdUDEbHi6FZ__U8EOQw

This should be modeled as a string too.

Reply:
@ryansbenson We are from Azure data factory team. Our team has discussed about this and make an agreement that not to block this kind of changes for now till ARM check with the Azure Architecture team and get a different decision. This is indeed a by-design in ADF at moment, not a breaking change. @ravbhatnagar is familiar for our scenario.

BTW, just give a little more context on ADF as well:

Regarding to ADF business needs, it is an data integration service which support 80+ data stores and certain number of other compute system as well, we model these data stores as LinkedService (with a “type” property to identify the actual data store type), and the each piece of data in these data stores as DataSet (with a “type” property to identify the actual data type), while we have another model object called Pipeline which refer to LinkedServices and DataSets to define activities to process these data. As you can see, in this model design, whenever we add a new type of LinkedService and DataSet or adding new properties to a particular LinkedService and DataSet, it doesn’t impact the impact the main concept of our model and it doesn’t mean a major change to ADF services, so we don’t want to take this kind of changes as major changes and upgrade the API version . In the other hand, if we always upgrade API version in this case, it will be very difficult for customer to figure out which API version support what data stores given we have large number of  data stores and we add this data store support very quickly.

Thanks
Xiaoying

Reply:
@ryansbenson  Do you have any  further question about this change?

Reply:
My original comment still holds true, this is a breaking change to ARM. I understand ADF would like to disregard this requirement but we can't let anyone break things on purpose.

It seems your team needs to address your impression of a breaking change. The same feedback is left here too:
https://github.com/Azure/azure-rest-api-specs/pull/7947